lib/chef/provisioning/chef_provider_action_handler.rb

```diff
@@ -42,16 +42,23 @@ def report_progress(description)
       provider.converge_by description do
         # We already did the action, but we trust whoever told us that they did it.
       end
+      puts "\n"
     end
 
     def performed_action(description)
       provider.converge_by description do
         # We already did the action, but we trust whoever told us that they did it.
       end
+      puts "\n"
```

Comment:
@jkeiser I agree, `resource_updated` is only supposed to be called _after_ all the converge_bys complete.  If provider actions (the chef provisioning concept) only take place _inside_ providers then when the provider/resource pair exits, it will correctly un-indent the formatter and add a trailing newline.

So I think you are right - `converge_by` is the correct method to call, not `run_action`.

I would actually expect that core chef would have this same issue - `Chef::Log` lines mixed within event formatter lines would appear on the same line.  Aaaaaaaand I just read your comment below which says the same thing :smile: 

Some other ideas:

1. Force `Chef::Log` output to a file instead of stdout so it does not clutter the event formatter output.
1. Update the logging system in chef core to ensure log lines start at a new line, and if not, it adds one.

Thoughts?